### DISCUSSION TITLE
Phone number input field format

### DISCUSSION THREAD

**Discussion by ishah-twilio** [twilio-labs/paste#695] [Q&A] [answered]
Hello

We're looking to upgrading SIM order page to paste component. One of the required filed is "country code + phone number".
As of today, we don't show supporting help text that shows user the format to enter the phone number. We like to auto-format user entered string. Parsing the inputted value to the format of the phone number supported by the specific country.

In error scenario we like to give out meaningful error that explains user the phone number format specific to the selected country code.

Is there existing library or paste component that we can use to auto format phone number input field? Any concerns we need to address for accessibility if we are proposing the auto formatting the input filed.

![phone number input format](https://user-images.githubusercontent.com/69016180/91360462-59308f80-e7ab-11ea-8ede-4f6434082a07.png)

**Comment by kate-porter** [comment]
@richbachman @SiTaggart can our inputs support auto-formatting?

**Comment by aayushpi** [comment, marked as answer]
No. We leave formatting of fields to the delivery team since they can provide the best possible experience. That being said, I would speak to your partner PDs in Phone Numbers to see if they have come up with any ideas. @vnguyen94 or @ramoneguru perhaps?

**Reply by vankhoawin** [reply]
We currently don't have any component like that in Phone Numbers. This does look like [Cleave](https://nosir.github.io/cleave.js/) would help with creating a good phone number input experience.

Recently I've seen more requests about larger Twilio-shared React components. I think this requires a larger discussion for Twilio frontend folks. Maybe next webguild meeting? :)

**Reply by ghostling** [reply]
We're actually planning on using [libphonenumber-js](https://www.npmjs.com/package/libphonenumber-js) for the type ahead functionality since we need it for validation anyway.

**Reply by aayushpi** [reply]
@vnguyen94 would you be interested in taking a spot and moderating it? I think that's a great call.

**Reply by vankhoawin** [reply]
talked offline with Aayush, we'll be discussing this topic at the next web guild meeting

**Comment by richbachman** [comment]
@ishah-twilio I think a library like [Formik](https://formik.org/) would probably work for your needs. @vnguyen94 built a [Paste/Formik integration](https://github.com/vnguyen94/formik-paste) library that might be helpful as well.

**Reply by vankhoawin** [reply]
actually now lives in https://github.com/twilio-labs/formik-paste :) coming soon...
